plugins/scenario/World.ts

```diff
@@ -84,7 +84,8 @@ export class World {
   }
 
   async impersonateAddress(address: string, value?: bigint): Promise<SignerWithAddress> {
-    return impersonateAddress(this.deploymentManager, address, value);
+    const dm = this.auxiliaryDeploymentManager ?? this.deploymentManager;
```

Comment:
this change makes me nervous; things start behaving strangely when you impersonate an address with the wrong deployment manager.

I got stuck for a long time yesterday because I was impersonating an address with the wrong deployment manager. transactions that I knew should be reverting were executing successfully.

swapping out this value seems likely to cause difficult debugging issues in the future

Reply:
Fair, but then should we just pass a flag to make it explicit?

Reply:
our just use the underyling `impersonateAddress` function directly